src/App.tsx

```diff
@@ -15,48 +15,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type ReorderOptions = {
+  reverse: boolean,
+  type: number,
+};
+
+function ReorderGoods(goods: string[], { reverse, type }: ReorderOptions) {
+  const initialGoods = [...goods];
+
+  switch (type) {
+    case 1:
+      initialGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case 2:
+      initialGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (reverse) {
+    initialGoods.reverse();
+  }
+
+  return initialGoods;
+}
+
 export const App: React.FC = () => {
+  const [type, useType] = useState(0);
```

Comment:
instead number create and use sortType enum.
it will increase readibility and decrease possibilty potential errors